File: 3658-gcd-of-odd-and-even-sums/3658-gcd-of-odd-and-even-sums.cpp
class Solution {
public:
    int gcdOfOddEvenSums(int n) {
        // return n;
        int e = 0 ;
        int o = 0;

        for(int num = 1 ; num <= 2*n  ; num++){
            if(num%2 == 0){
                e += num;
            }
            else{
                o += num;
            }
        }

        return g(e , o);
    }
private:
    int g(int a , int b){
        if(a > b) return g(a-b , b);
        else if (b > a) return g(a , b -a);
        return b;
    }
};

File: 3658-gcd-of-odd-and-even-sums/3658-gcd-of-odd-and-even-sums.py
class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        # return n can do it but for god sake ! 
        e = 0
        o = 0

        for num in range(1 , 2*n + 1):
            if num % 2 == 0 :
                e += num 
            else :
                o += num
        
        def g(a , b):
            if a > b: return g(a - b, b)
            elif b > a: return g(a, b - a)
            return b

        return g(e , o)